tests/testsuite/test_acceptance_with_browsers.py

```diff
@@ -121,10 +122,20 @@ def wait_for_reconfigure():
 @pytest.fixture(params=BROWSERS)
 def browser(request):
     browser_name = request.param
-    with splinter.Browser(browser_name, headless=True, wait_time=15, **BROWSER_OPTS[browser_name]) as browser:
-        browser.driver.set_window_size(1920, 1080)
-        browser.visit(URL)
-        yield browser
+    for _ in range(3):
+        try:
+            with splinter.Browser(browser_name,
+                                  headless=True,
+                                  wait_time=15,
+                                  **BROWSER_OPTS[browser_name]) as browser:
+                browser.driver.set_window_size(1920, 1080)
+                browser.visit(URL)
+                yield browser
+                break
+        except WebDriverException:
+            continue
+    else:
+        pytest.fail("Got WebDriverException 3 times in a row")
```

Comment:
I think the 3rd time we should re-raise the exception, and i'm really not a fan of the indentation on line 127. How about this?

```python
    retry = 3
    while retry:
        retry -= 1
        try:
            browser = splinter.Browser(
                browser_name,
                headless=True,
                wait_time=15,
                **BROWSER_OPTS[browser_name]
            )
            browser.driver.set_window_size(1920, 1080)
            browser.visit(URL)
            yield browser
            return
        except WebDriverException:
            if retry:
                continue
            else:
                raise
```

Reply:
we still need the `with` there, so it calls `close` on the browser driver